### PR TITLE
[WNMGDS-1025] Fix spanish link navigation

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary

WNMGDS-1025

Hgov header and footer have a link with the same text (Spanish). I added `aria-label` attrs to the links to give better context to what a screen reader user is seeing.

To test, open Stories for the [Header](https://cmsgov.github.io/design-system/branch/WNMGDS-1025/BUGFIX-spanish-link-navigation/storybook/?path=/story/healthcare-header--logged-out-header&globals=theme:healthcare) ~and [Footer](https://cmsgov.github.io/design-system/branch/WNMGDS-1025/BUGFIX-spanish-link-navigation/storybook/?path=/story/healthcare-footer--basic-footer&globals=theme:healthcare)~ component. On the language links you should see `aria-label` text in dev tools giving more context to the link.

~All language links in the footer should have this label.~

Only the Spanish link in the header should have this label. The login/logout link shouldn't.

**NOTE:** I'm not fluent in Spanish and used Google translate to get the Spanish equivalent of the English aria-label text. Might need someone to confirm what I added makes sense.

### CODE REVIEW

File: packages/ds-healthcare-gov/src/components/Header/__snapshots__/defaultMenuLinks.test.js.snap [hunk 1]
```diff
@@ -12,6 +12,7 @@ Object {
       "label": "My profile",
     },
     Object {
+      "ariaLabel": "CuidadoDeSalud.gov en Español",
```
Comment (zarahzachz) [inline packages/ds-healthcare-gov/src/components/Header/__snapshots__/defaultMenuLinks.test.js.snap:15]: I'm a little curious what this snapshot test is actually testing

Comment (zarahzachz) [reply packages/ds-healthcare-gov/src/components/Header/__snapshots__/defaultMenuLinks.test.js.snap:15]: 🤔 Looks like a lot of snapshot tests around hgov Header and Footer look like this.

File: packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx [hunk 2]
```diff
@@ -70,7 +70,11 @@ const InlineLinkLists = function (props: InlineLinkListsProps) {
         {Object.getOwnPropertyNames(languages).map(function (lang, index) {
           return (
             <li key={lang} className={inlineLiClasses}>
-              <a lang={lang} href={primaryDomain + languages[lang].href}>
+              <a
+                lang={lang}
+                aria-label={`${languages[lang].label} ${props.t('footer.languageResources')}`}
```
[the remark this answers is not in the document]
Comment (zarahzachz) [reply packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx:75]: Do we have a ticket in place to follow-up with content people? Afaik this is the only ticket documenting this problem.

[the remark this answers is not in the document]
Comment (pwolfert) [reply packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx:75]: As far as I know, we only got the go-ahead to update the header. We'll need to follow up with Product

[the remark this answers is not in the document]
Comment (zarahzachz) [reply packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx:75]: I commented on the Jira ticket and tagged Krithi

**Review by pwolfert — APPROVED** [review]
Works great 🎉

**Pull request merged**